### PR TITLE
SCP-1687: adjust WalletServer logging

### PR DESCRIPTION
**Pull request by gilligan**
**Description**: Switch logging in `Cardano.Wallet` to freer logging

- Added `WalletMsg` type for the structural logging
- Massaging the types for Wallet.Server.main
- Very minor code refactoring in `Wallet.Server.main`

----

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-pab/app/Main.hs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE NamedFieldPuns        #-}
 {-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE RecordWildCards       #-}
```
Comment (michaelpj) [inline plutus-pab/app/Main.hs:11]: Controversial :p I don't mind them... but maybe we need a style guide for the odd extensions we do object to.

File: plutus-pab/src/Cardano/Wallet/Server.hs [hunk 2]
```diff
@@ -98,25 +98,24 @@ app nodeClientEnv chainIndexEnv mVarState =
...
-            $ flip handleError (error . show @ClientError)
-            $ ChainIndexClient.handleChainIndexClient chainIndexEnv
-            $ startWatching (Wallet.ownAddress state)
-    let warpSettings :: Warp.Settings
-        warpSettings = Warp.defaultSettings & Warp.setPort port & Warp.setBeforeMainLoop (available availability)
-    logInfoN $ "Starting wallet server on port: " <> tshow port
+    runClient chainIndexEnv
+    logInfo $ StartingWallet servicePort
     liftIO $ Warp.runSettings warpSettings $ app nodeClientEnv chainIndexEnv mVarState
+    where
```
Comment (michaelpj) [inline plutus-pab/src/Cardano/Wallet/Server.hs:109]: So one style thing that I think we *do* usually go for is to prefer define-before-use, i.e. `let` over `where` unless it's a really big thing.

I am a bit atypical in my anti-where prejudices, though :)

File: plutus-pab/src/Cardano/Wallet/Types.hs [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
+import           Cardano.BM.Data.Tracer        (ToObject (..))
+import           Cardano.BM.Data.Tracer.Extras (Tagged (..), mkObjectStr)
+import           Data.Aeson                    (FromJSON, ToJSON)
+import           Data.Text                     (Text)
+import           Data.Text.Prettyprint.Doc     (Pretty (..), (<+>))
+import           GHC.Generics                  (Generic)
+import           Servant.Client                (BaseUrl)
+import           Wallet.Emulator.Wallet        (Wallet)
 
+type NodeUrl = BaseUrl
```
Comment (michaelpj) [inline plutus-pab/src/Cardano/Wallet/Types.hs:21]: Hmm, maybe we should newtype these...

File: plutus-pab/src/Cardano/Wallet/Server.hs [hunk 4]
```diff
@@ -98,25 +98,24 @@ app nodeClientEnv chainIndexEnv mVarState =
...
-        state = emptyWalletState wallet
-    nodeClientEnv <-
-        liftIO $ do
-            nodeManager <- newManager defaultManagerSettings
-            pure $ mkClientEnv nodeManager nodeBaseUrl
-    chainIndexEnv <-
-        liftIO $ do
-            chainIndexManager <- newManager defaultManagerSettings
-            pure $ mkClientEnv chainIndexManager chainIndexBaseUrl
+main :: Trace IO WalletMsg -> Config -> NodeUrl -> ChainIndexUrl -> Availability -> IO ()
```
Comment (michaelpj) [inline plutus-pab/src/Cardano/Wallet/Server.hs:101]: Hmm, not sure what our policy should be on `MonadIO` vs `IO`, e.g. there are some other `MonadIO` things in this module. I'd be fine with mostly fixing IO, I think?

**Review by michaelpj — APPROVED** [review]
Looks fine, just some style things (all optional, I think).

File: plutus-pab/app/Main.hs (hunk 1 above)
Comment (gilligan) [reply plutus-pab/app/Main.hs:11]: I think that would be a good idea, yes!

File: plutus-pab/src/Cardano/Wallet/Server.hs (hunk 4 above)
Comment (gilligan) [reply plutus-pab/src/Cardano/Wallet/Server.hs:101]: yeah sounds good to me. I'll do that

Comment (gilligan) [reply plutus-pab/src/Cardano/Wallet/Server.hs:101]: Actually - on second thought: i want to keep these PRs "minimally invasive' while i get to know the code base better. I do note all your comments though and *will* get back to addressing them! Thanks!

**Pull request merged**